pd/src/state.rs

```diff
@@ -496,18 +498,27 @@ ON CONFLICT (id) DO UPDATE SET data = $1
     }
 
     /// Retrieve the [`Asset`] for a given asset ID.
-    pub async fn asset_lookup(&self, asset_id: Vec<u8>) -> Result<Asset> {
+    pub async fn asset_lookup(&self, asset_id: Vec<u8>) -> Result<params::AssetInfo> {
         let mut conn = self.pool.acquire().await?;
 
         let asset = query!(
-            "SELECT denom, asset_id FROM assets WHERE asset_id = $1",
+            "SELECT denom, asset_id, total_supply FROM assets WHERE asset_id = $1",
             asset_id
         )
         .fetch_one(&mut conn)
         .await?;
-        Ok(Asset {
-            asset_denom: asset.denom,
-            asset_id: asset.asset_id,
+
+        // TODO: should we be returning proto types from our state methods, or domain types?
+        Ok(params::AssetInfo {
+            denom: Some(
+                asset::REGISTRY
+                    .parse_denom(asset.denom.as_str())
+                    .unwrap()
+                    .into(),
+            ),
+            asset_id: Some(asset::Id::try_from(asset.asset_id)?.into()),
+            total_supply: asset.total_supply.try_into()?, // postgres only has i64....
```

Comment:
I think it'd be better to do `as u64` here, so that in the (unlikely) event that we'd recorded a `total_supply > 2^63`, we'd get the correct value out (since the `u64 <> i64` `as` conversion is lossless, just not semantically-preserving, and we don't do arithmetic inside the database).